insights/client/__init__.py

```diff
@@ -306,23 +296,13 @@ def verify(self, egg_path, gpg_key=constants.pub_gpg_path):
                     'rc': 0}
 
         # if a valid egg path and gpg were received do the verification
-        if egg_path and gpg_key:
-            cmd_template = '/usr/bin/gpg --verify --keyring %s %s %s'
-            cmd = cmd_template % (gpg_key, egg_path + '.asc', egg_path)
-            logger.debug(cmd)
-            process = Popen(shlex.split(cmd), stdout=PIPE, stderr=PIPE)
-            stdout, stderr = process.communicate()
-            rc = process.returncode
-            logger.debug("GPG return code: %s" % rc)
-            return {'gpg': True if rc == 0 else False,
-                    'stderr': stderr,
-                    'stdout': stdout,
-                    'rc': rc}
-        else:
-            return {'gpg': False,
-                    'stderr': 'Must specify a valid core and gpg key.',
-                    'stdout': 'Must specify a valid core and gpg key.',
-                    'rc': 1}
+        if egg_path:
+            return crypto.verify_gpg_signed_file(file=egg_path, signature=egg_path + ".asc")
```

Comment:
You can filter for specific commits in the **Files changed** tab. I prefer not introducing dumb merge conflicts when the work can be expressed as several consecutive (yet self-contained) changes.

Reply:
It felt small enough. I'll split it into smaller commits.

Reply:
I would rather prefer multiple smaller pull requests, not because I want to nitpick, but for the pure sake of practicality: the diffs are clearer and it is easier to review the patch. Also, when there are multiple changes, having them in separate PRs allows not only to comment on them separately, but also to merge the indepentent parts faster. Yes, there may be merge conflicts, but that’s nothing to avoid or be afraid of.